core/src/main/java/feast/core/service/AccessManagementService.java

```diff
@@ -60,6 +63,8 @@ public void archiveProject(String name) {
     Optional<Project> project = projectRepository.findById(name);
     if (!project.isPresent()) {
       throw new IllegalArgumentException(String.format("Could not find project: \"%s\"", name));
+    } else if (name.equals(Project.DEFAULT_NAME)) {
+      throw new IllegalArgumentException("Archiving the default project is not allowed.");
```

Comment:
Nitpick, but can we use guard clauses? https://refactoring.com/catalog/replaceNestedConditionalWithGuardClauses.html

Reply:
Rectified

Reply:
Corrected to `if (name.equals(Project.DEFAULT_NAME)) {`